### PR TITLE
Fix NPE when Vertex AI listSessions omits sessions

### PR DESCRIPTION
**Pull request by rolandkakonyi**
Fixes a `NullPointerException` when the Vertex AI Session API listSessions response omits `sessions` or returns `sessions: null`.

Problem:
```
java.lang.NullPointerException: Cannot invoke "java.util.List.iterator()" because "apiSessions" is null
  at com.google.adk.sessions.VertexAiSessionService.parseListSessionsResponse ( com/google.adk.sessions/VertexAiSessionService.java:133 )
  at com.google.adk.sessions.VertexAiSessionService.lambda$listSessions$0 ( com/google.adk.sessions/VertexAiSessionService.java:121 )
  at io.reactivex.rxjava3.internal.operators.maybe.MaybeMap$MapMaybeObserver.onSuccess ( io/reactivex.rxjava3.internal.operators.maybe/MaybeMap.java:83 )
  at io.reactivex.rxjava3.internal.operators.single.SingleFlatMapMaybe$FlatMapMaybeObserver.onSuccess ( io/reactivex.rxjava3.internal.operators.single/SingleFlatMapMaybe.java:114 )
  at io.reactivex.rxjava3.internal.operators.maybe.MaybeJust.subscribeActual ( io/reactivex.rxjava3.internal.operators.maybe/MaybeJust.java:36 )
  at io.reactivex.rxjava3.core.Maybe.subscribe ( io/reactivex.rxjava3.core/Maybe.java:5377 )
  at io.reactivex.rxjava3.internal.operators.single.SingleFlatMapMaybe$FlatMapSingleObserver.onSuccess ( io/reactivex.rxjava3.internal.operators.single/SingleFlatMapMaybe.java:86 )
  at io.reactivex.rxjava3.internal.operators.single.SingleFromCallable.subscribeActual ( io/reactivex.rxjava3.internal.operators.single/SingleFromCallable.java:55 )
  at io.reactivex.rxjava3.core.Single.subscribe ( io/reactivex.rxjava3.core/Single.java:4855 )
  at io.reactivex.rxjava3.internal.operators.single.SingleFlatMapMaybe.subscribeActual ( io/reactivex.rxjava3.internal.operators.single/SingleFlatMapMaybe.java:38 )
  at io.reactivex.rxjava3.core.Maybe.subscribe ( io/reactivex.rxjava3.core/Maybe.java:5377 )
  at io.reactivex.rxjava3.internal.operators.maybe.MaybeMap.subscribeActual ( io/reactivex.rxjava3.internal.operators.maybe/MaybeMap.java:41 )
  at io.reactivex.rxjava3.core.Maybe.subscribe ( io/reactivex.rxjava3.core/Maybe.java:5377 )
  at io.reactivex.rxjava3.internal.operators.maybe.MaybeToSingle.subscribeActual ( io/reactivex.rxjava3.internal.operators.maybe/MaybeToSingle.java:46 )
  at io.reactivex.rxjava3.core.Single.subscribe ( io/reactivex.rxjava3.core/Single.java:4855 )
  at io.reactivex.rxjava3.core.Single.blockingGet ( io/reactivex.rxjava3.core/Single.java:3644 )
  at com.google.adk.web.controller.SessionController.listSessions ( com/google.adk.web.controller/SessionController.java:136 )
  at jakarta.servlet.http.HttpServlet.service ( jakarta/servlet.http/HttpServlet.java:564 )
  at jakarta.servlet.http.HttpServlet.service ( jakarta/servlet.http/HttpServlet.java:658 )
```

Changes:
- Treat missing/null/empty `sessions` as an empty list in `VertexAiSessionService`.
- Add regression tests for missing + null `sessions`.

Test:
- `./mvnw -pl core test`

### CODE REVIEW

File: core/src/main/java/com/google/adk/sessions/VertexAiSessionService.java [hunk 1]
```diff
@@ -119,15 +119,20 @@ public Single<ListSessionsResponse> listSessions(String appName, String userId)
...
+      return ListSessionsResponse.builder().sessions(new ArrayList<>()).build();
+    }
     List<Map<String, Object>> apiSessions =
-        objectMapper.convertValue(
-            listSessionsResponseMap.get("sessions"),
-            new TypeReference<List<Map<String, Object>>>() {});
+        objectMapper.convertValue(sessionsNode, new TypeReference<List<Map<String, Object>>>() {});
+    if (apiSessions == null) {
+      apiSessions = new ArrayList<>();
+    }
```
[the remark this answers is not in the document]
Comment (rolandkakonyi) [reply core/src/main/java/com/google/adk/sessions/VertexAiSessionService.java]: Fixed in d0cd877ba1d171e1fbe26bda41c5bd108b992c70

File: core/src/test/java/com/google/adk/sessions/VertexAiSessionServiceTest.java [hunk 2]
```diff
@@ -322,6 +326,42 @@ public void listSessions_empty() {
...
+        .thenReturn(
+            new ApiResponse() {
+              @Override
+              public ResponseBody getResponseBody() {
+                return ResponseBody.create(JSON_MEDIA_TYPE, "{}");
+              }
+
+              @Override
+              public void close() {}
+            });
```
[the remark this answers is not in the document]
Comment (rolandkakonyi) [reply core/src/test/java/com/google/adk/sessions/VertexAiSessionServiceTest.java]: Fixed in 89b3ed77c32c99be3da4378ca087ecf0091cf93f

**Review by sherryfox — APPROVED** [review]

File: core/src/main/java/com/google/adk/sessions/VertexAiSessionService.java [hunk 3]
```diff
@@ -128,15 +128,17 @@ public Single<ListSessionsResponse> listSessions(String appName, String userId)
         .map(
             listSessionsResponseMap ->
                 parseListSessionsResponse(listSessionsResponseMap, appName, userId))
-        .defaultIfEmpty(ListSessionsResponse.builder().build());
+        .defaultIfEmpty(ListSessionsResponse.builder().sessions(new ArrayList<>()).build());
```
Comment (sherryfox) [inline core/src/main/java/com/google/adk/sessions/VertexAiSessionService.java:131]: The `builder()` already initializes the session with an empty list so we can probably skip this here and in line _138_.

File: core/src/test/java/com/google/adk/sessions/VertexAiSessionServiceTest.java [hunk 4]
```diff
@@ -319,6 +335,24 @@ public void listSessions_empty() {
...
+        .thenReturn(apiResponseJson("{}"));
+
+    assertThat(vertexAiSessionService.listSessions("123", "userX").blockingGet().sessions())
+        .isEmpty();
+  }
+
+  @Test
+  public void listSessions_nullSessionsField_returnsEmpty() {
+    when(mockApiClient.request("GET", "reasoningEngines/123/sessions?filter=user_id=userY", ""))
+        .thenReturn(apiResponseJson("{\"sessions\": null}"));
```
Comment (sherryfox) [inline core/src/test/java/com/google/adk/sessions/VertexAiSessionServiceTest.java:350]: WDYT about updating this to something like:
`.thenAnswer(new MockApiAnswer(...))` and adjusting `MockApiAnswer` implementation to respond with given rawApiResponse?

This way we could avoid importing a new lib (and using a non-trivial, realistic media type to mock ApiResponse) for the purpose of those tests

**Comment by rolandkakonyi** [comment]
> Hi @rolandkakonyi,
> 
> Thank you very much for your contribution! We really appreciate you taking the time and effort to submit this pull request.
> 
> To fully integrate the changes and ensure alignment with our project's internal guidelines and best practices, we've made some further adjustments in a follow-up.
> 
> Thanks again for your valuable contribution!

Thank you @sherryfox!

I'm sorry that I missed the notifications about the comments here.
Thanks for taking this over the finish line!
